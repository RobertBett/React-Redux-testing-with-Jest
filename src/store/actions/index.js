import { SAVE_COMMENT, FETCH_COMMENTS } from "./types";
import axios from "axios";


export const saveComment= (comment) =>({
    type: SAVE_COMMENT,
    payload: comment 
})

export const fetchComments =() =>{
    const response = axios.get('http://jsonplaceholder.typicode.com/comments')
    console.log(response)
    return{
        type: FETCH_COMMENTS,
        payload: response
    }
}
