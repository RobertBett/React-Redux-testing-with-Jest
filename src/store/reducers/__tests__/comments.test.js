import commentsReducer from '../comments';
import { SAVE_COMMENT } from '../../actions/types';
import toJSON  from 'enzyme-to-json';

describe('reducer component', () =>{
    let newState;
    it('handles actions of type SAVE_COMMENT', ()=> {
    const action ={
        type: SAVE_COMMENT,
        payload: 'New Comment'
    };
    newState = commentsReducer([], action);
    expect(newState).toEqual(['New Comment'])
    });
    it('handles action with uknown type', () => {
    newState = commentsReducer([], { type: 'vmkl;mvermvpoe'} );
    expect(newState).toEqual([])
    })
    it('should match the snapshot', () =>{
        expect(toJSON(newState)).toMatchSnapshot()
    })
})