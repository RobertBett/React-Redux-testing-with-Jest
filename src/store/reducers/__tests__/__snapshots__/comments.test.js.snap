// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer component should match the snapshot 1`] = `undefined`;
